### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by JcBernack**
- refactor `BigInteger` usages:
  - `compareTo(BigInteger.Zero)` => `signum()` same result, better performance
  - `longValue()` => `longValueExact()` make sure we throw an exception instead of working with unexpected values
  - `intValue()` => `intValueExact()` same
- refactor base fee calculation
  - add more test cases to make sure the edge cases are covered
- cleanup RPC methods

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala [hunk 1]
```diff
@@ -403,36 +376,18 @@ class EthService(
...
-  ) = {
-    implicit val timeout: Timeout = 5 seconds
-    val barrier = Await
-      .result(sidechainTransactionActorRef ? BroadcastTransaction(transaction), timeout.duration)
-      .asInstanceOf[Future[Unit]]
-    onComplete(barrier) {
-      // TODO: add correct responses
-      case Success(_) => ApiResponseUtil.toResponse(transactionResponseRepresentation(transaction))
-      case Failure(exp) =>
-        ApiResponseUtil.toResponse(GenericTransactionError("GenericTransactionError", JOptional.of(exp)))
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala:422]: It seems that we don't cover this case anymore

File: sdk/src/main/scala/com/horizen/account/utils/AccountFeePaymentsUtils.scala [hunk 2]
```diff
@@ -49,7 +49,8 @@ object AccountFeePaymentsUtils {
         // sum all rewards for this forger address
         val forgerTotalFee = allForgersRewards
           .filter(info => forgerKey.equals(info.address))
-          .foldLeft(BigInteger.ZERO)((sum, info) => sum.add(info.value))
+          .map(_.value)
+          .reduce(_.add(_))
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/utils/AccountFeePaymentsUtils.scala:53]: does it work faster for us?
honestly, for me foldLeft approach was more clear.

File: sdk/src/test/scala/com/horizen/account/state/AccountStateTest.scala [hunk 3]
```diff
@@ -87,7 +87,7 @@ class AccountStateTest
     feePayments = state.getFeePayments(0)
     assertEquals(s"Fee payments size expected to be different.", 3, feePayments.size)
 
-    var forgerTotalFee = feePayments.foldLeft(BigInteger.ZERO)((sum, payment) => sum.add(payment.value))
+    var forgerTotalFee = feePayments.map(_.value).reduce(_.add(_))
```
Comment (i-Alex) [inline sdk/src/test/scala/com/horizen/account/state/AccountStateTest.scala:90]: same here

File: sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala [hunk 4]
```diff
@@ -109,7 +109,7 @@ class ForgerStakeMsgProcessorTest
     if (!view.accountExists(origin)) {
       view.addAccount(origin, randomHash)
 
-      if (amount.compareTo(BigInteger.ZERO) >= 0) {
+      if (amount.signum() == 1) {
```
Comment (i-Alex) [inline sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala:112]: good, no sense to increase the balance in case of `0`

File: sdk/src/test/scala/com/horizen/account/utils/FeeUtilsTest.scala [hunk 5]
```diff
@@ -1,24 +1,49 @@
...
+      currentBaseFee: Long,
+      expectedBaseFee: Long
+  ): Unit = {
+    val block =
+      AccountMockDataHelper(false).getMockedBlock(BigInteger.valueOf(currentBaseFee), gasUsedPercent, 100, null, null)
+    assertEquals(message, calculateNextBaseFee(block), BigInteger.valueOf(expectedBaseFee))
+  }
+
+  /**
+   * Separate test for calculate next base fee as it is not used in base fee validator test
```
Comment (i-Alex) [inline sdk/src/test/scala/com/horizen/account/utils/FeeUtilsTest.scala:24]: looks much better

File: sdk/src/main/scala/com/horizen/account/utils/AccountFeePaymentsUtils.scala (hunk 2 above)
Comment (JcBernack) [reply sdk/src/main/scala/com/horizen/account/utils/AccountFeePaymentsUtils.scala:53]: I would have preferred `map(_.value).sum`, but that doesn't work with the Java BigInteger, unfortunately. It would work with the Scala BigInt but we're not using that anywhere yet. Using `reduce` is still the better option I think, because it allows the List implementation to sum the elements in any order, potentially using a faster algorithm.

File: sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala (hunk 1 above)
Comment (JcBernack) [reply sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala:422]: - If I'm not mistaken `ApiResponseUtil.toResponse()` is the wrong tool to use within the RPC api as it would write to the response directly, bypassing our RPC response serialization.
- Also I expected `Await.result` to throw in case the future does not resolve - and exceptions are caught further up and returned as an RPC error response. Did not test this though.

File: sdk/src/main/scala/com/horizen/account/utils/AccountFeePaymentsUtils.scala (hunk 2 above)
Comment (JcBernack) [reply sdk/src/main/scala/com/horizen/account/utils/AccountFeePaymentsUtils.scala]: done

File: sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala (hunk 1 above)
Comment (JcBernack) [reply sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala:422]: Fixed and added a simple python test that checks that an invalid TX is rejected with an error, instead of getting a success response with the tx hash.

**Review by i-Alex — APPROVED** [review]
